Source/Libraries/GSF.Web/Hosting/WebPageController.cs

```diff
@@ -235,31 +286,75 @@ public IEnumerable<object> GetServices(Type serviceType)
+            public AuthenticationOptions AuthenticationOptions { get; set; }
+            public Action<HttpRouteCollection> RouteConfig { get; set; } = DefaultRouteConfig;
+
+            public IWebPageControllerBuilder UseDefaultWebPage(string defaultWebPage)
+            {
+                DefaultWebPage = defaultWebPage;
+                return this;
+            }
+
+            public IWebPageControllerBuilder UseModel<T>(T model) =>
+                UseModel(model, typeof(T));
+
+            public IWebPageControllerBuilder UseModel(object model, Type modelType)
+            {
+                Model = model;
+                ModelType = modelType;
+                return this;
+            }
+
+            public IWebPageControllerBuilder UseAuthenticationOptions(AuthenticationOptions authenticationOptions)
+            {
+                AuthenticationOptions = authenticationOptions;
+                return this;
+            }
+
+            public IWebPageControllerBuilder UseCustomRoutes(Action<HttpRouteCollection> routeConfig)
+            {
+                RouteConfig = routeConfig ?? DefaultRouteConfig;
+                return this;
+            }
+
+            private static void DefaultRouteConfig(HttpRouteCollection _) { }
+        }
+
         /// <summary>
         /// Registers web page controller in web server pipeline.
         /// </summary>
         /// <param name="app">The app builder for the web server pipeline.</param>
         /// <param name="webServer"><see cref="WebServer"/> instance to use for controller.</param>
-        /// <param name="defaultWebPage">The default page to display on the default path.</param>
-        /// <param name="model">Reference to model to use when rendering Razor templates, if any.</param>
-        /// <param name="modelType">Type of <paramref name="model"/>, if any.</param>
-        /// <param name="options">Authentication options for enabling sessions.</param>
-        public static void UseWebPageController(this IAppBuilder app, WebServer webServer, string defaultWebPage = "Index.html", object model = null, Type modelType = null, AuthenticationOptions options = null)
+        /// <param name="webPageControllerConfig">Function used to configure the web page controller.</param>
+        public static void UseWebPageController(this IAppBuilder app, WebServer webServer, Action<IWebPageControllerBuilder> webPageControllerConfig)
         {
+            WebPageControllerBuilder builder = new WebPageControllerBuilder();
+            webPageControllerConfig(builder);
+
             HttpConfiguration httpConfig = new HttpConfiguration();
-            httpConfig.DependencyResolver = GetDependencyResolver(webServer, model, modelType);
+            httpConfig.DependencyResolver = GetDependencyResolver(webServer, builder.Model, builder.ModelType);
+
+            AuthenticationOptions options = builder.AuthenticationOptions;
 
-            if (options != null)
+            if (!(options is null))
```

Comment:
FYI, the expression `if (options is not null)` should be supported now with latest C# version - all libraries in GSF have been updated to use this.

Reply:
```suggestion
            if (options is not null)
```